test/functional/core/job_spec.lua

```diff
@@ -303,6 +303,26 @@ describe('jobs', function()
     ]])
   end)
 
+  it('requires funcrefs for script-local (s:) functions', function()
```

Comment:
Hmm, then shouldn't we be able to translate the `s:foo` to something that works without requiring `function('s:foo')`?  We should have `scriptID` available at that time.